SmartTutor/Database/SmartTutorContext.cs

```diff
@@ -100,5 +108,21 @@ private static void ConfigureBasicMetricChecker(ModelBuilder modelBuilder)
                 .WithOne()
                 .HasForeignKey("MethodMetricCheckerForeignKey");
         }
+
+        private static void ConfigureProjectChecker(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<ChallengeFulfillmentStrategy>()
+               .Property<string?>("StrategiesApplicableToSnippetCheckerForeignKey").IsRequired(false);
+
+            modelBuilder.Entity<ProjectChecker>()
+                .Property(pc => pc.StrategiesApplicableToSnippet)
+                .IsRequired()
+                .HasConversion<string[]>(
+                    sats => ((IEnumerable)sats).Cast<object>().Select(x => x.ToString()).ToArray(),
+                    sats => sats == null
+                        ? new Dictionary<string, List<ChallengeFulfillmentStrategy>>()
+                        : JsonConvert.DeserializeObject<Dictionary<string, List<ChallengeFulfillmentStrategy>>>(sats.ToString())
```

Comment:
Instead of introducing a new dependency (Netwonsoft), we should use the built in JSON serializer (https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-5-0).

In general, we want to depend on the least amount of external code possible. While it may be convenient to plug in a new library with a few clicks there is a price that we pay for maintaining a link to that library. Ideally, we can do our job with the built in functionality, as is the case with the System.Text.Json package.